cluster_test.go

```diff
@@ -1693,20 +1693,47 @@ func (s *S) TestMaxSocketUses(c *C) {
 		time.Sleep(5e8)
 	}
 
-	// a socket was used during connection
-	numUses := 1
+	for i := 0; i < 2; i++ {
+		stats = mgo.GetStats()
+		// every 4th usage should result in a new connection
+		c.Logf("Active connections: %d", stats.SocketsAlive)
+		s := session.Copy()
+		defer s.Close()
+		c.Check(s.Ping(), IsNil)
+		time.Sleep(time.Second)
```

Comment:
cgo is is an external library and it talks to actual mongo instance running locally (not just one instance, it launches 15 instances of mongo) and isnt cleanly designed to mock objects out. So as much as I hate sleep I'm force to use it here.